File: cleanup_geojson.js
const fs = require('fs');

// Properties to remove
const propertiesToRemove = ["ST", "DT", "TS_PCODE", "TS_MMR"];

// Load the GeoJSON file
fs.readFile('township_live copy.geojson', 'utf8', (err, data) => {
  if (err) {
    console.error('Error reading file:', err);
    return;
  }

  try {
    const geojsonData = JSON.parse(data);

    // Remove properties from all features
    geojsonData.features.forEach(feature => {
      if (feature.properties) {
        propertiesToRemove.forEach(prop => {
          if (prop in feature.properties) {
            delete feature.properties[prop];
          }
        });
      }
    });

    // Save the cleaned data to a new file
    fs.writeFile('township_live_cleaned.geojson', JSON.stringify(geojsonData, null, 2), 'utf8', (err) => {
      if (err) {
        console.error('Error writing file:', err);
        return;
      }
      console.log('Cleanup complete! Removed properties: ST, DT, TS_PCODE, TS_MMR');
      console.log('Cleaned file saved as: township_live_cleaned.geojson');
    });
  } catch (parseErr) {
    console.error('Error parsing JSON:', parseErr);
  }
});

File: cleanup_geojson.py
import json

# Load the GeoJSON file
with open('township_live.geojson', 'r', encoding='utf-8') as f:
    geojson_data = json.load(f)

# Properties to remove
properties_to_remove = ["ST", "DT", "TS_PCODE", "TS_MMR"]

# Remove properties from all features
for feature in geojson_data['features']:
    if 'properties' in feature:
        for prop in properties_to_remove:
            if prop in feature['properties']:
                del feature['properties'][prop]

# Save the cleaned data to a new file
with open('township_live_cleaned.geojson', 'w', encoding='utf-8') as f:
    json.dump(geojson_data, f, ensure_ascii=False, indent=2)

print("Cleanup complete! Removed properties: ST, DT, TS_PCODE, TS_MMR")
print("Cleaned file saved as: township_live_cleaned.geojson")
